src/components/latest-rules-content/latestRulesContent.js

```diff
@@ -69,23 +70,77 @@ const LatestRulesContent = ({
     return rule.slice(6, rule.length);
   };
 
+  const openUserRule = async (path) => {
+    const loginName = await fetchGithubName(path);
+    const pathPrefix = process.env.NODE_ENV === 'development' ? '' : '/rules';
+
+    if (loginName) {
+      navigate(`${pathPrefix}/user-rules/?author=${loginName}`);
+    } else {
+      navigate('/404');
+    }
+  };
+
+  const fetchGithubName = async (path) => {
+    const githubOwner = process.env.GITHUB_ORG;
+    const githubRepo = process.env.GITHUB_REPO;
+    const token = process.env.GITHUB_API_PAT;
+    const response = await fetch(
+      `https://api.github.com/repos/${githubOwner}/${githubRepo}/commits?path=${path}`,
+      {
+        headers: {
+          Authorization: `bearer ${token}`,
+        },
+      }
+    );
+
+    if (!response.ok) {
+      throw new Error(
+        `GitHub API request failed with status: ${response.status}`
+      );
+    }
+
+    const data = await response.json();
+    const loginName = data[data.length - 1]?.author?.login;
+
+    return loginName;
+  };
+
```

Comment:
Can we get this information from gatsby or cosmos db? 
I feel like we store this information somewhere?

Reply:
Unfortunately, in Gatsby and CosmosDB, we only have this type of data. For now, we still need to use the GitHub API to retrieve people's login names. 😣

![image](https://github.com/SSWConsulting/SSW.Rules/assets/127192800/a017f9d0-f4cf-4368-9a07-12eaecc89986)

Reply:
ah damn.... cool this looks good then 🙂